src/main/java/core/basesyntax/service/impl/FileReaderImpl.java

```
@@ -0,0 +1,18 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+public class FileReaderImpl implements FileReader {
+    @Override
+    public List<String> read(String filePath) {
+        try {
+            return Files.readAllLines(Paths.get(filePath));
```

Either use try-with-resources with a BufferedReader (Files.newBufferedReader(Paths.get(filePath))) or keep Files.readAllLines but still validate inputs and improve the error message. Per checklist: "try-with-resources" and "Don't ignore exceptions in try-catch blocks... Let's rethrow a RuntimeException with an informative message and exception object." If you keep Files.readAllLines, ensure the catch message matches the suggested style (for example: "Can't find file by path: " + filePath) and include the original exception as the cause (you already include the cause).